pkg/steps/source.go

```diff
@@ -243,10 +244,31 @@ func handleBuild(buildClient BuildClient, build *buildapi.Build, dry bool, artif
 
 			if isInfraReason(b.Status.Reason) {
 				log.Printf("Build %s previously failed from an infrastructure error (%s), retrying...\n", b.Name, b.Status.Reason)
-				opts := &meta.DeleteOptions{Preconditions: &meta.Preconditions{UID: &b.UID}}
+				zero := int64(0)
+				foreground := meta.DeletePropagationForeground
```

Comment:
foreground should not return until you've deleted... but retry is fine.

Reply:
Just gotta be sure yaknow?